### PR TITLE
chore: Update datafusion, arrow/flight/parquet to `46.0.0` , object_store to `0.7.0`

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
* get https://github.com/apache/arrow-rs/pull/4646, closes https://github.com/influxdata/idpe/issues/17930 and one of the last known sources of OOM kills in the compactor / querier.
* Keep up with dependencies

# Changes:
1. ~Update DataFusion pin to https://github.com/apache/arrow-datafusion/commit/aa1d72c10f372989e28b928db447558ded66a670~
2. Update DataFusion pin to https://github.com/apache/arrow-datafusion/commit/3ea870cf0995a86b5297a92be5d5432577aefabe
3. Update arrow/parquet/flight to a patched version of `46.0.0` https://github.com/apache/arrow-rs/pull/4790 (to get the fix for https://github.com/apache/arrow-rs/issues/4788)

### CODE REVIEW

File: iox_query/src/exec.rs [hunk 1]
```diff
@@ -712,6 +730,10 @@ mod tests {
             .chain(futures::stream::once(async move {
                 barrier.wait().await;
                 Ok(RecordBatch::new_empty(schema))
+            }))
+            .chain(futures::stream::once(async move {
```
Comment (alamb) [inline iox_query/src/exec.rs:734]: This test had a race condition for me / the task scheduling was different -- specifically SortExec would complete before the test had a chance to verify the memory use. Thus I put in a second barrier to ensure it worked correctly

File: iox_query_influxql/src/window/non_negative.rs [hunk 2]
```diff
@@ -35,6 +38,8 @@ impl PartitionEvaluator for NonNegative {
         num_rows: usize,
     ) -> Result<Arc<dyn Array>> {
         let array = self.partition_evaluator.evaluate_all(values, num_rows)?;
+        // see https://github.com/apache/arrow-rs/issues/4781
+        #[allow(deprecated)]
         let predicate = lt_dyn_scalar(&array, 0)?;
```
Comment (alamb) [inline iox_query_influxql/src/window/non_negative.rs:43]: See // see https://github.com/apache/arrow-rs/issues/4781 (I am not sure how to update this interface)

File: parquet_file/src/storage.rs [hunk 3]
```diff
@@ -137,7 +137,7 @@ impl ParquetExecInput {
             limit: None,
             table_partition_cols: vec![],
             // Parquet files ARE actually sorted but we don't care here since we just construct a `collect` plan.
-            output_ordering: vec![vec![]],
```
Comment (alamb) [inline parquet_file/src/storage.rs:140]: The `vec![vec![]]` was causing https://github.com/apache/arrow-datafusion/issues/7482

File: querier/src/cache/object_store.rs [hunk 4]
```diff
@@ -27,29 +27,60 @@ use super::ram::RamSize;
 
 const CACHE_ID: &str = "object_store";
 
+#[derive(Debug, Clone)]
```
Comment (alamb) [inline querier/src/cache/object_store.rs:30]: Now that the GetResult must also return `ObjectMeta` I updated this PR to cache the metadata as well. Assuming that this is a reasonable thing to do I will file a follow on ticket in arrow to allow better calculation of memory use

File: querier/src/cache/object_store.rs [hunk 5]
```diff
@@ -290,16 +322,8 @@ impl ObjectStore for CachedObjectStore {
     }
 
     async fn head(&self, location: &Path) -> Result<ObjectMeta, ObjectStoreError> {
-        let data = self.get_data(location).await?;
-
-        Ok(ObjectMeta {
-            location: location.clone(),
-            // nobody really cares about the "last modified" field and it is wasteful to issue a HEAD request just to
```
Comment (alamb) [inline querier/src/cache/object_store.rs:297]: Now that the result contains the `ObjectMeta` I return the correct value here

File: object_store_metrics/src/lib.rs [hunk 6]
```diff
@@ -228,29 +228,36 @@ impl ObjectStore for ObjectStoreMetrics {
         let res = self.inner.get_opts(location, options).await;
 
         match res {
-            Ok(GetResult::File(file, path)) => {
-                // Record the file size in bytes and time the inner call took.
-                if let Ok(m) = file.metadata() {
-                    self.get_bytes.inc(m.len());
-                    if let Some(d) = self.time_provider.now().checked_duration_since(started_at) {
-                        self.get_success_duration.record(d)
+            Ok(mut res) => {
```
Comment (alamb) [inline object_store_metrics/src/lib.rs:231]: The object store API changed non trivially: https://github.com/apache/arrow-rs/pull/4677

File: compactor/tests/integration.rs [hunk 7]
```diff
@@ -233,7 +233,7 @@ async fn test_compact_large_overlapes() {
     - "L2.5[136000,136000] 300s 2kb                                                                                          |L2.5|"
     - "L2.6[6000,30000] 240s 3kb|-----L2.6-----|                                                                          "
     - "L2.7[36000,36000] 240s 3kb                    |L2.7|                                                                "
-    - "L2.8[68000,68000] 240s 2kb                                          |L2.8|                                          "
+    - "L2.8[68000,68000] 240s 3kb                                          |L2.8|                                          "
```
Comment (alamb) [inline compactor/tests/integration.rs:236]: I don't know why these tests change (perhaps a small difference in the parquet metadata has resulted in their being sightly different sizes)

File: service_common/src/error.rs [hunk 8]
```diff
@@ -27,6 +27,7 @@ pub fn datafusion_error_to_tonic_code(e: &DataFusionError) -> tonic::Code {
         DataFusionError::ResourcesExhausted(_) => tonic::Code::ResourceExhausted,
         // Map as many as possible back into user visible (non internal) errors
         DataFusionError::SQL(_)
+        | DataFusionError::Configuration(_)
```
Comment (crepererum) [inline service_common/src/error.rs:30]: Shouldn't this map to "internal" (from a users PoV)?

File: querier/src/cache/object_store.rs (hunk 4 above)
Comment (crepererum) [reply querier/src/cache/object_store.rs:30]: sounds reasonable

File: iox_query/src/exec.rs (hunk 1 above)
Comment (crepererum) [reply iox_query/src/exec.rs:734]: I have to admit that I've seen this issue in some CI runs but decided to ignore it til it bugs me enough :sweat_smile:

**Review by crepererum — COMMENTED** [review]
Looks like there's an arrow bug:


```text
2023-09-05T20:50:11.195582Z DEBUG iox_query::provider::deduplicate: done sending the left over batch
   0: rust_begin_unwind
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panicking.rs:593:5
   1: core::panicking::panic_fmt
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/panicking.rs:67:14
   2: core::panicking::panic
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/panicking.rs:117:5
   3: core::option::Option<T>::unwrap
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/option.rs:935:21
   4: arrow_array::array::byte_array::GenericByteArray<T>::value_unchecked
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-array-46.0.0/src/array/byte_array.rs:294:13
   5: <&arrow_array::array::byte_array::GenericByteArray<T> as arrow_ord::cmp::ArrayOrd>::value_unchecked
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-ord-46.0.0/src/cmp.rs:522:9
   6: arrow_ord::cmp::apply_op_vectored::{{closure}}
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-ord-46.0.0/src/cmp.rs:450:12
   7: arrow_ord::cmp::collect_bool
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-ord-46.0.0/src/cmp.rs:388:24
   8: arrow_ord::cmp::apply_op_vectored
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-ord-46.0.0/src/cmp.rs:447:5
   9: arrow_ord::cmp::apply
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-ord-46.0.0/src/cmp.rs:326:17
  10: arrow_ord::cmp::compare_op::{{closure}}
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-ord-46.0.0/src/cmp.rs:219:29
  11: arrow_ord::cmp::compare_op
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-ord-46.0.0/src/cmp.rs:286:44
  12: arrow_ord::cmp::neq
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-ord-46.0.0/src/cmp.rs:86:5
  13: iox_query::exec::seriesset::converter::SeriesSetConverter::compute_transitions
             at /home/rust/project/iox_query/src/exec/seriesset/converter.rs:186:14
  14: iox_query::exec::seriesset::converter::SeriesSetConverter::compute_changepoints::{{closure}}
             at /home/rust/project/iox_query/src/exec/seriesset/converter.rs:142:25
  15: core::iter::adapters::map::map_fold::{{closure}}
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/iter/adapters/map.rs:84:28
  16: <core::slice::iter::Iter<T> as core::iter::traits::iterator::Iterator>::fold
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/slice/iter/macros.rs:215:27
  17: <core::iter::adapters::map::Map<I,F> as core::iter::traits::iterator::Iterator>::fold
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/iter/adapters/map.rs:124:9
  18: core::iter::traits::iterator::Iterator::for_each
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/iter/traits/iterator.rs:856:9
  19: alloc::vec::Vec<T,A>::extend_trusted
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/alloc/src/vec/mod.rs:2828:17
  20: <alloc::vec::Vec<T,A> as alloc::vec::spec_extend::SpecExtend<T,I>>::spec_extend
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/alloc/src/vec/spec_extend.rs:26:9
  21: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter_nested::SpecFromIterNested<T,I>>::from_iter
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/alloc/src/vec/spec_from_iter_nested.rs:62:9
  22: <alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/alloc/src/vec/spec_from_iter.rs:33:9
  23: <alloc::vec::Vec<T> as core::iter::traits::collect::FromIterator<T>>::from_iter
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/alloc/src/vec/mod.rs:2696:9
  24: core::iter::traits::iterator::Iterator::collect
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/iter/traits/iterator.rs:1895:9
  25: iox_query::exec::seriesset::converter::SeriesSetConverter::compute_changepoints
             at /home/rust/project/iox_query/src/exec/seriesset/converter.rs:140:31
  26: <iox_query::exec::seriesset::converter::SeriesSetConverterStream as futures_core::stream::Stream>::poll_next
             at /home/rust/project/iox_query/src/exec/seriesset/converter.rs:398:36
  27: <core::pin::Pin<P> as futures_core::stream::Stream>::poll_next
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-core-0.3.28/src/stream.rs:120:9
  28: futures_util::stream::stream::StreamExt::poll_next_unpin
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-util-0.3.28/src/stream/stream/mod.rs:1632:9
  29: <futures_util::stream::stream::next::Next<St> as core::future::future::Future>::poll
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-util-0.3.28/src/stream/stream/next.rs:32:9
  30: iox_query::exec::cross_rt_stream::CrossRtStream<core::result::Result<X,E>>::new_with_error_stream::{{closure}}::{{closure}}
             at /home/rust/project/iox_query/src/exec/cross_rt_stream.rs:89:53
  31: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::future::future::Future>::poll
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/panic/unwind_safe.rs:296:9
  32: <futures_util::future::future::catch_unwind::CatchUnwind<Fut> as core::future::future::Future>::poll::{{closure}}
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-util-0.3.28/src/future/future/catch_unwind.rs:36:42
  33: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/panic/unwind_safe.rs:271:9
  34: std::panicking::try::do_call
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panicking.rs:500:40
  35: __rust_try
  36: std::panicking::try
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panicking.rs:464:19
  37: std::panic::catch_unwind
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panic.rs:142:14
  38: <futures_util::future::future::catch_unwind::CatchUnwind<Fut> as core::future::future::Future>::poll
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-util-0.3.28/src/future/future/catch_unwind.rs:36:9
  39: executor::DedicatedExecutor::spawn::{{closure}}
             at /home/rust/project/executor/src/lib.rs:324:69
  40: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/future/future.rs:125:9
  41: executor::Task::run::{{closure}}::{{closure}}
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/macros/select.rs:524:49
  42: <tokio::future::poll_fn::PollFn<F> as core::future::future::Future>::poll
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/future/poll_fn.rs:58:9
  43: executor::Task::run::{{closure}}
             at /home/rust/project/executor/src/lib.rs:65:9
  44: executor::DedicatedExecutor::new_inner::{{closure}}::{{closure}}::{{closure}}
             at /home/rust/project/executor/src/lib.rs:271:40
  45: <tracing::instrument::Instrumented<T> as core::future::future::Future>::poll
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tracing-0.1.37/src/instrument.rs:272:9
  46: tokio::runtime::task::core::Core<T,S>::poll::{{closure}}
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/core.rs:334:17
  47: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/loom/std/unsafe_cell.rs:16:9
  48: tokio::runtime::task::core::Core<T,S>::poll
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/core.rs:323:13
  49: tokio::runtime::task::harness::poll_future::{{closure}}
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/harness.rs:485:19
  50: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/panic/unwind_safe.rs:271:9
  51: std::panicking::try::do_call
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panicking.rs:500:40
  52: __rust_try
  53: std::panicking::try
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panicking.rs:464:19
  54: std::panic::catch_unwind
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panic.rs:142:14
  55: tokio::runtime::task::harness::poll_future
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/harness.rs:473:18
  56: tokio::runtime::task::harness::Harness<T,S>::poll_inner
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/harness.rs:208:27
  57: tokio::runtime::task::harness::Harness<T,S>::poll
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/harness.rs:153:15
  58: tokio::runtime::task::raw::poll
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/raw.rs:276:5
  59: tokio::runtime::task::raw::RawTask::poll
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/raw.rs:200:18
  60: tokio::runtime::task::LocalNotified<S>::run
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/mod.rs:400:9
  61: tokio::runtime::scheduler::multi_thread::worker::Context::run_task::{{closure}}
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/scheduler/multi_thread/worker.rs:639:17
  62: tokio::runtime::coop::with_budget
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/coop.rs:107:5
  63: tokio::runtime::coop::budget
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/coop.rs:73:5
  64: tokio::runtime::scheduler::multi_thread::worker::Context::run_task
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/scheduler/multi_thread/worker.rs:575:9
  65: tokio::runtime::scheduler::multi_thread::worker::Context::run
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/scheduler/multi_thread/worker.rs:526:24
  66: tokio::runtime::scheduler::multi_thread::worker::run::{{closure}}::{{closure}}
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/scheduler/multi_thread/worker.rs:491:21
  67: tokio::runtime::context::scoped::Scoped<T>::set
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/context/scoped.rs:40:9
  68: tokio::runtime::context::set_scheduler::{{closure}}
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/context.rs:176:26
  69: std::thread::local::LocalKey<T>::try_with
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/thread/local.rs:270:16
  70: std::thread::local::LocalKey<T>::with
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/thread/local.rs:246:9
  71: tokio::runtime::context::set_scheduler
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/context.rs:176:9
  72: tokio::runtime::scheduler::multi_thread::worker::run::{{closure}}
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/scheduler/multi_thread/worker.rs:486:9
  73: tokio::runtime::context::runtime::enter_runtime
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/context/runtime.rs:65:16
  74: tokio::runtime::scheduler::multi_thread::worker::run
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/scheduler/multi_thread/worker.rs:478:5
  75: tokio::runtime::scheduler::multi_thread::worker::Launch::launch::{{closure}}
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/scheduler/multi_thread/worker.rs:447:45
  76: <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/blocking/task.rs:42:21
  77: <tracing::instrument::Instrumented<T> as core::future::future::Future>::poll
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tracing-0.1.37/src/instrument.rs:272:9
  78: tokio::runtime::task::core::Core<T,S>::poll::{{closure}}
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/core.rs:334:17
  79: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/loom/std/unsafe_cell.rs:16:9
  80: tokio::runtime::task::core::Core<T,S>::poll
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/core.rs:323:13
  81: tokio::runtime::task::harness::poll_future::{{closure}}
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/harness.rs:485:19
  82: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/panic/unwind_safe.rs:271:9
  83: std::panicking::try::do_call
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panicking.rs:500:40
  84: __rust_try
  85: std::panicking::try
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panicking.rs:464:19
  86: std::panic::catch_unwind
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panic.rs:142:14
  87: tokio::runtime::task::harness::poll_future
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/harness.rs:473:18
  88: tokio::runtime::task::harness::Harness<T,S>::poll_inner
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/harness.rs:208:27
  89: tokio::runtime::task::harness::Harness<T,S>::poll
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/harness.rs:153:15
  90: tokio::runtime::task::raw::poll
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/raw.rs:276:5
  91: tokio::runtime::task::raw::RawTask::poll
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/raw.rs:200:18
  92: tokio::runtime::task::UnownedTask<S>::run
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/task/mod.rs:437:9
  93: tokio::runtime::blocking::pool::Task::run
             at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/runtime/blocking/pool.rs:159:9
```

**Comment by alamb** [comment]
I am currently chasing down a panic

Reproducer:


`input.lp`:

```text
cpu_load_short,host=server01,region=us-west value=0.64 1694013632168976000
cpu_load_short,host=server01 value=27.99 1694013632168976001
cpu_load_short,host=server02,region=us-west value=3.89 1694013632168976002
cpu_load_short,host=server01,region=us-east value=1234567.891011 1694013632168976003
cpu_load_short,host=server01,region=us-west value=0.000003 1694013632168976004
system,host=server03 uptime=1303385i 1694013632168976005
swap,host=server01,name=disk0 in=3i,out=4i 1694013632168976006
status active=true 1694013632168976007
attributes color="blue" 1694013632168976008
```

The run like this
```shell
influxdb_iox write 1099926508025198_9276191685857031 /tmp/input.lp
influxdb_iox storage 1099926508025198_9276191685857031 read-filter --start 1694013632168976000 --stop 1694013632168976010 --predicate="host='server01'"
```

File: service_common/src/error.rs (hunk 8 above)
Comment (alamb) [reply service_common/src/error.rs:30]: Yes, I think that is a good call given the places in the code that can throw a `DataFusionError::Configuration`. I will make the change

https://github.com/search?q=repo%3Aapache%2Farrow-datafusion+Configuration+language%3ARust&type=code&l=Rust

**Comment by alamb** [comment]
I am working on extracting a self contained reproducer

**Comment by alamb** [comment]
> Looks like there's an arrow bug:

Filed upstream issue:  https://github.com/apache/arrow-rs/issues/4788

Once we have a fix for that I'll plan to pin this PR to patched `46.0.0` branch and get this PR in

File: service_common/src/error.rs (hunk 8 above)
Comment (alamb) [reply service_common/src/error.rs]: in 285cb74aa

File: iox_query_influxql/src/window/non_negative.rs (hunk 2 above)
Comment (alamb) [reply iox_query_influxql/src/window/non_negative.rs]: fixed in da10871a4

**Review by crepererum — APPROVED** [review]
:shipit:

**Comment by alamb** [comment]
I plan to merge this tomorrow morning, as soon as the current incident has settled down and I have the maxmimum overlap time to observe the rollout

**Comment by alamb** [comment]
Slack announcement; https://influxdata.slack.com/archives/CRK9M8L5Q/p1694171211622649

**Comment by alamb** [comment]
Revert PR prepared in case: https://github.com/influxdata/influxdb_iox/pull/8696

File: querier/src/cache/object_store.rs (hunk 4 above)
Comment (alamb) [reply querier/src/cache/object_store.rs:30]: I filed  https://github.com/apache/arrow-rs/issues/4798 upstream and made https://github.com/influxdata/influxdb_iox/pull/8697 to add a reference in the code.

**Pull request merged**